java-cfenv-boot-pivotal-sso/src/main/java/io/pivotal/cfenv/boot/sso/CfSingleSignOnLegacyProcessor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pivotal.cfenv.boot.sso;
+
+import java.util.Map;
+
+import io.pivotal.cfenv.core.CfCredentials;
+import io.pivotal.cfenv.core.CfService;
+import io.pivotal.cfenv.spring.boot.CfEnvProcessor;
+import io.pivotal.cfenv.spring.boot.CfEnvProcessorProperties;
+
+/**
+ * @author Mark Pollack
+ * @author Scott Frederick
+ * @author Pivotal Application Single Sign-On
+ */
+public class CfSingleSignOnLegacyProcessor implements CfEnvProcessor {
+
+	private static final String PIVOTAL_SSO_LABEL = "p-identity";
+
+	@Override
+	public boolean accept(CfService service) {
+		return service.existsByLabelStartsWith(PIVOTAL_SSO_LABEL);
```

Comment:
I believe this PR will end up invoking both the `CfSingleSignOnLegacyProcessor` and the updated `CfSingleSignOnProcessor` which will create confusion in terms of which boot properties are set in the running application.  Using a Class.forName test on an oauth class that i only in the legacy support and another test for the new spring security 5 oauth support should be done in the accept method.

Reply:
Hi @markpollack. Per our discussion on Slack, we have reintroduced the `Class.forName` (presumably `ClassUtils.isPresent` uses `Class.forName` under the hood?) changes back into the PR. 

There is one corner case that we wanted to surface:
If an application using this library somehow has both versions of Spring Security loaded, or more specifically the `org.springframework.security.oauth2.common.DefaultOAuth2AccessToken` AND the `org.springframework.security.core.Authentication` class, then both Processors will return true for their `accept` methods and set their Spring properties. We're unsure exactly what the behavior would be in their client application if this were the case, however it would be the same behavior as when we always ran both Processors and the those multiple Spring Security version classes actually existed. If you're ok with this risk, then so are we.